pylint/checkers/strings.py

```diff
@@ -541,15 +549,45 @@ class StringConstantChecker(BaseTokenChecker):
     # Unicode strings.
     UNICODE_ESCAPE_CHARACTERS = 'uUN'
 
+    def __init__(self, *args, **kwargs):
+        super(StringConstantChecker, self).__init__(*args, **kwargs)
+        self.string_tokens = {}
+
     def process_module(self, module):
         self._unicode_literals = 'unicode_literals' in module.future_imports
 
     def process_tokens(self, tokens):
-        for (tok_type, token, (start_row, _), _, _) in tokens:
+        for (tok_type, token, start, _, _) in tokens:
             if tok_type == tokenize.STRING:
                 # 'token' is the whole un-parsed token; we can look at the start
                 # of it to see whether it's a raw or unicode string etc.
-                self.process_string_token(token, start_row)
+                self.process_string_token(token, start[0])
+                try:
+                    self.string_tokens[start] = ast.literal_eval(token)
+                except ValueError:
+                    pass
+
+    @check_messages(*(MSGS.keys()))
+    def visit_list(self, node):
+        self.check_for_concatenated_strings(node, 'list')
+
+    @check_messages(*(MSGS.keys()))
+    def visit_set(self, node):
+        self.check_for_concatenated_strings(node, 'set')
+
+    @check_messages(*(MSGS.keys()))
+    def visit_tuple(self, node):
+        self.check_for_concatenated_strings(node, 'tuple')
+
+    def check_for_concatenated_strings(self, iterable_node, iterable_type):
+        for elt in iterable_node.elts:
+            if isinstance(elt, Const) and elt.pytype() in _AST_NODE_STR_TYPES:
```

Comment:
You can probably check the type of `elt.value` instead of using `pytype`

Reply:
I'm not sure to understand how, sorry

The 2nd member of this conditional serves to distinguish between e.g. string constants and integer constants.
Are you suggesting to do something like the collowing code ?
```
if isinstance(elt, Const) and isinstance(elt.value, six.string_types):
```